google/gax/__init__.py

```diff
@@ -482,3 +489,160 @@ def __next__(self):
         if self._index >= len(self._current):
             self._current = None
         return resource
+
+
+def _from_any(pb_type, any_pb):
+    if not any_pb.Is(pb_type.DESCRIPTOR):
+        raise TypeError(
+            'Could not convert {} to {}'.format(
+                any_pb.__class__.__name__, pb_type.__name__))
+
+    return pb_type.FromString(any_pb.value)
+
+
+class ResultError(GaxError):
```

Comment:
I disagree. :-) I think there could be a case where a developer wants to catch result errors but does not want to catch other ways that a `GaxError` might be thrown, now or in the future. The subclass makes this really easy, and is much cleaner than catching `GaxError`, checking `cause`, and then re-raising if you do not recognize it.